src/qtils/error.hpp

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright Quadrivium LLC
+ * All Rights Reserved
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <any>
+#include <cstdint>
+#include <deque>
+#include <system_error>
+
+#include <fmt/format.h>
+
+#include <qtils/optref.hpp>
+
+#define Q_ERROR_LOCATION \
+  (::qtils::ErrorLocation{__PRETTY_FUNCTION__, __FILE__, __LINE__})
+
+#define Q_ERROR(error_) (::qtils::Error{Q_ERROR_LOCATION, error_})
+
+namespace qtils::error {
+  struct ErrorLocation {
+    const char *fn;
+    const char *file;
```

Comment:
string_view?

Reply:
Don't want to increase `sizeof(ErrorLocation)`